### PR TITLE
replay game when entered final room

### PR DESCRIPTION
**Pull request by zhaoan12**
when stepping into final room, replay game

### CODE REVIEW

**Review by m-grigoriu — APPROVED** [review]

**Comment by 6167656e74323431** [comment]
It is unclear when the game ends... it just feels like the game goes in an infinite loop (also the minimap does not get cleared and so it just keeps growing:

e.g.
image

**Review by 6167656e74323431 — CHANGES_REQUESTED** [review]
Isn't there supposed to be a reward and some kind of indication when you beat the game? It's incredibly confusing when that happens currently, and afaik there isn't any reward.

Edit: we should probably also heal when you win.

File: src/main/java/com/mg105/interface_adapters/WinDisplay.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.mg105.interface_adapters;
+
+import com.mg105.use_cases.RoomGetter;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+public class WinDisplay implements PropertyChangeListener {
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/interface_adapters/WinDisplay.java:8]: missing javadoc

File: src/main/java/com/mg105/interface_adapters/WinDisplay.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private final Toggler toggler;
+    private final RoomGetter roomGetter;
+
+    public WinDisplay(Toggler toggler, RoomGetter roomGetter){
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/interface_adapters/WinDisplay.java:12]: missing javadoc

File: src/main/java/com/mg105/user_interface/WinMenu.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.mg105.user_interface;
+
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import org.jetbrains.annotations.NotNull;
+
+public class WinMenu implements Toggleable {
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/user_interface/WinMenu.java:9]: missing javadoc

File: src/main/java/com/mg105/user_interface/WinMenu.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private final @NotNull Scene scene;
+    private final @NotNull Pane layout;
+
+    public WinMenu(@NotNull ReplayGeneratorButton ReplayButton) {
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/user_interface/WinMenu.java:13]: missing javadoc

File: src/main/java/com/mg105/user_interface/WinMenu.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        Button generateMapButton = new Button("You Won! Replay The Game");
+        //Clean Inventory
+        generateMapButton.setOnAction(ReplayButton);
+        layout = new StackPane();
+        layout.getChildren().add(generateMapButton);
+        scene = new Scene(layout, 600, 600);
+    }
+
+    @Override
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/user_interface/WinMenu.java:22]: missing javadoc

File: src/main/java/com/mg105/user_interface/WinMenu.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    public @NotNull Scene getScene() {
+        return scene;
+    }
+
+    @Override
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/user_interface/WinMenu.java:27]: missing javadoc

**Review by 6167656e74323431 — CHANGES_REQUESTED** [review]

**Review by 6167656e74323431 — APPROVED** [review]

**Review by siddharthgowda — APPROVED** [review]

**Review by m-grigoriu — APPROVED** [review]
Looks good.

**Pull request merged**